sds_data_manager/lambda_code/SDSCode/path_helper.py

```diff
@@ -0,0 +1,186 @@
+    """This class stores filename pattern configuration."""
+
+    mission: str = "imap"
+    instruments: list = field(
+        default_factory=lambda: [
+            "codice",
+            "glows",
+            "hit",
+            "hi-45",
+            "idex",
+            "lo",
+            "mag",
+            "swapi",
+            "swe",
+            "ultra-45",
+        ]
+    )
+    data_level: list = field(
+        default_factory=lambda: ["l0", "l1", "l1a", "l1b", "l1c", "l1d", "l2"]
+    )
+    descriptor: str = None
+    startdate: str = "YYYYMMDD"
+    enddate: str = "YYYYMMDD"
+    version: str = r"^v\d{2}-\d{2}$"
+    extensions: list = field(default_factory=lambda: ["pkts", "cdf"])
+
+
+class FilenameParser:
+    def __init__(self, filename):
+        """Stores common methods to check
+        file patterns and build upload path.
+
+        Current filename convention:
+        <mission>_<instrument>_<datalevel>_<descriptor>_<startdate>_<enddate>_<version>.<extension>
+
+        NOTE: There are no optional parameters anymore. All parameters are required.
+        <mission>: imap
+        <instrument>: idex, swe, swapi, hi-45, ultra-45 and etc.
+        <datalevel> : l1a, l1b, l1, l3a and etc.
+        <descriptor>: descriptor stores information specific to instrument. This is
+            decided by each instrument.
+        <startdate>: startdate is the earliest date in the data. Format - YYYYMMDD
+        <enddate>: Some instrument and some data level requires to store date range.
+            If there is no end date, then startdate will be used as enddate as well.
+            Format - YYYYMMDD.
+        <version>: This stores software version and data version. Version format is
+            vxx-xx.
+
+        Parameters
+        ----------
+        filename : str
+            Filename
+        """
+        self.filename_convention = (
+            "<mission>_<instrument>_<datalevel>_<descriptor>_"
+            "<startdate>_<enddate>_<version>.<extension>"
+        )
+        self.split_filename = filename.replace("_", ".").split(".")
+        (
+            self.mission,
+            self.instrument,
+            self.data_level,
+            self.descriptor,
+            self.startdate,
+            self.enddate,
+            self.version,
+            self.extension,
+        ) = self.split_filename
+        # This message is returned to user through API to indicate why filename was not
+        # correct.
+        self.message = None
+
+    def check_date_input(self, input_date: str) -> bool:
+        """Check input date string is in valid format and is correct date
+
+        Parameters
+        ----------
+        input_date : str
+            Date in YYYYMMDD format.
+
+        Returns
+        -------
+        bool
+            Whether date input is valid or not
+        """
+        # Check if the pattern matches 8 digits (YYYYMMDD)
+        if not re.match(r"^\d{8}$", input_date):
+            return False
```

Comment:
Do you need this, or can you just let your try/except below catch this case too?

Reply:
Agree with Greg, below will catch only 8 digit times